crates/aptos-protos/proto/aptos/transaction/v1/transaction.proto

```diff
@@ -0,0 +1,415 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package aptos.transaction.v1;
+
+import "aptos/util/timestamp/timestamp.proto";
+
+// A block on Aptos holds transactions in chronological order (ordered by a transactions monotonically increasing `version` field)
+// All blocks start with a `BlockMetadataTransaction`, and are followed by zero or more transactions.
+// The next `BlockMetadataTransaction` denotes the end of the current block, and the start of the next one.
+//
+// The Block `height` is a strictly monotonically increasing count of the number of blocks,
+// and there will never be a gap in the numbers. It is also a unique identifier: there will never be two blocks with
+// the same `height`.
+//
+// The Genesis Transaction (version 0) is contained within the first block, which has a height of `0`
+message Block {
+  // Timestamp represents the timestamp of the `BlockMetadataTransaction` (or `GenesisTransaction` for the genesis block)
+  // and every transaction in the `transactions` will have the same `timestamp` as the block.
+  aptos.util.timestamp.Timestamp timestamp = 1;
+
+  // Height represents the block number and ultimately, is the count of `BlockMetadataTransaction` that happened on the chain.
+  uint64 height = 2;
+
+  // Transactions holds all transactions that happened in the Block, which is transactions that happened starting with (and including)
+  // a `BlockMetadataTransaction`, and every other transaction up to (but excluding) the next `BlockMetadataTransaction`.
+  repeated Transaction transactions = 3;
+
+  // Chain ID informs us which chain we're trying to index, this is important to ensure that we're not mixing chains within a single pipeline.
+  uint32 chain_id = 4;
+}
+
+// Transaction as it happened on the chain, there are 4 types of transactions:
+// - User Transaction: a user initiated transaction to interact with the chain
+// - Block Metadata Transaction: transactions generated by the chain to group together transactions forming a "block"
+// - State Checkpoint Transaction: transactions generated by the chain so when validator agreed on a particular global state
+// - Genesis Transaction: the first transaction of the chain, with all core contract and validator information baked in
+message Transaction {
+  aptos.util.timestamp.Timestamp timestamp = 1;
+  uint64 version = 2;
+  TransactionInfo info = 3;
+  uint64 epoch = 4;
+  uint64 block_height = 5;
+
+  enum TransactionType {
+    GENESIS = 0;
```

Comment:
Could we make the 0 item in all of these enum as NOT_SET please. That way we know for sure whether it's the default value or was explicitly set. If we don't do this we'll probably regret it.

Reply:
Ah interesting. Let me see how complicated it gets.